### ISSUE TITLE
WARNING-uvicorn.error httptools_impl 181 Unsupported upgrade request.

### ISSUE THREAD

**Issue by mtheory945846** [Kludex/uvicorn#954]
- [X] The bug is reproducible against the latest release and/or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

I am running a Starlette app with Uvicorn in a CentOS 7 / Python 3.8 Docker container. I am logging to files, and I keep seeing this log line repeat every couple of seconds:

```
2021-02-12 06:20:23,642-INFO-uvicorn.error-12::server|63:: Started server process [12]
2021-02-12 06:20:27,329-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:27,336-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:27,341-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,325-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,332-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,338-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,355-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,362-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,368-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:42,384-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
```

### To reproduce

Here are 2 different requirements.txt file contents I have tried while building and deploying my Docker Container to get this log to go away:

```
asyncio==3.4.3
aiofiles==0.6.0
uvloop==0.15.0
websockets==8.1
Cython==0.29.21
wsproto==1.0.0
httptools==0.1.1
h11==0.12.0
uvicorn==0.13.3
starlette==0.14.1
```
and
```
asyncio==3.4.3
aiofiles==0.6.0
starlette==0.14.1
uvicorn[standard]
```

### Expected behavior

I read [here](https://www.uvicorn.org/settings/):

```
--http <str> - Set the HTTP protocol implementation. The httptools implementation provides greater performance, but it not compatible with PyPy, and requires compilation on Windows. Options: 'auto', 'h11', 'httptools'. Default: 'auto'.
```

### Actual behavior

I am running a Starlette app with Uvicorn in a CentOS 7 / Python 3.8 Docker container. I am logging to files, and I keep seeing this log line repeat every couple of seconds:

```
2021-02-12 06:20:23,642-INFO-uvicorn.error-12::server|63:: Started server process [12]
2021-02-12 06:20:27,329-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:27,336-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:27,341-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,325-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,332-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:32,338-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,355-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,362-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:37,368-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
2021-02-12 06:20:42,384-WARNING-uvicorn.error-12::httptools_impl|181:: Unsupported upgrade request.
```

h11 also produces this repeated warning log. I can't get either http protocol implementation to run without these warnings.

### Debugging material

<img width="754" alt="Screen Shot 2021-02-12 at 2 08 24 AM" src="https://user-images.githubusercontent.com/70536960/107739705-7a445880-6cd7-11eb-8a97-9a47a778e205.png">

<img width="1416" alt="Screen Shot 2021-02-12 at 2 09 55 AM" src="https://user-images.githubusercontent.com/70536960/107739708-7adcef00-6cd7-11eb-8d2e-cb6ec4b4d98c.png">

### Environment

- Mac OSX
- Docker Container -> CentOS7 / Python3.8 / Uvicorn version: 0.13.3

I'm running:

CMD [ "uvicorn", "app:app", "--host", "0.0.0.0", "--port", "8003", "--debug", "--log-level", "debug", "--access-log", "--loop", "uvloop", "--ws", "auto", "--lifespan", "off", "--interface", "auto", "--forwarded-allow-ips", "*", "--limit-concurrency", "100", "--limit-max-requests", "100", "--backlog", "100", "--timeout-keep-alive", "5" ]

### Additional context

Related [issue](https://github.com/tiangolo/fastapi/issues/2209): (even though I'm not using FastAPI)

I don't want my log files filling up in production with this repeated warning. I really enjoy developing with Uvicorn so far before this.

**Comment by euri10** [comment]
tough to know without seeing your app, my best guess would be that you have a ws client trying to reach and most likely no ws route in your app, either stop calling a ws endpoint that doesn't exist or create it

either way you can filter logs in your logging settings (see an example [here](https://gist.github.com/wshayes/067fe781af341d4783e113f037c3959d))

**Comment by mtheory945846** [comment]
Thanks for your reply, @euri10. I am successfully running only the Starlette WebSocketEndpoint and HTTPEndpoint example from [here](https://www.starlette.io/endpoints/). Both endpoint types work and log to file for me. I don't know how uvicorn repeatedly uses these classes, [httptools_impl.py](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/httptools_impl.py) and [h11_impl.py](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/h11_impl.py), but I believe that the logs are coming from them without me doing anything else in code after startup.

**Comment by euri10** [comment]
> but I believe that the logs are coming from them without me doing anything else in code after startup.

can you provide the app you're running please ? can't help you without.

**Comment by mtheory945846** [comment]
I started over with a skeleton Starlette and Uvicorn app and now the warning log doesn't show. I slowly put my app source code into my skeleton app with repeated restarts. For a few hours, I thought using port 8003 was the culprit. I just ran my separate skeleton Starlette and Uvicorn app on port 8003 again, and it works fine without that log. I'm not sure how to reproduce this issue any longer.
